kroxylicious/src/main/java/io/kroxylicious/proxy/filter/KrpcFilterContext.java

```diff
@@ -27,6 +27,12 @@ public interface KrpcFilterContext {
      */
     ByteBuf allocate(int initialCapacity);
 
+    /**
+     * @return the SNI hostname provided by the client.  Will be null if the client is
+     * using a non-TLS connection or the TLS client hello didn't provide one.
+     */
+    String sniHostname();
```

Comment:
Related to the conversations in `Multitenant/MultiTenantTransformationFilter` I'm not sure this is the right level of abstraction. 

I guess there might be non multi-tenant use cases for the SNI hostname so maybe it should stay?

Reply:
My gut tells me that SNI hostname might have use beyond the world of multi-tenancy that will help make routing decisions.  I thinking of a use-case when a single kroxy pool acts a gateway for many kafka clusters.